src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe  
+    /// </summary>
+    public class EmbedAppNameInHost : TaskBase
+    {
+        private ITaskItem _outputAppHostPath;
+
+        [Required]
+        public string AppHostSourcePath { get; set; }
+
+        [Required]
+        public string AppHostDestinationDirectoryPath { get; set; }
+
+        [Required]
+        public string AppName { get; set; }
```

Comment:
It is not really AppName but AppBinaryName, right? Can we reflect it in the field naming as well?